framework/di/Instance.php

```diff
@@ -107,7 +107,9 @@ public static function of($id)
      */
     public static function ensure($reference, $type = null, $container = null)
     {
-        if ($reference instanceof $type) {
+        if (is_object($reference) && $type === null) {
+            return $reference;
+        } elseif (is_object($reference) && $reference instanceof $type) {
```

Comment:
if you remove the is_object leads to error `Error: Class name must be a valid object or a string`
so we checked is_string below, then here you need to check is_object